cmd/agentbasedinstaller/register.go

```diff
@@ -66,7 +66,14 @@ func RegisterCluster(ctx context.Context, log *log.Logger, bmInventory *client.A
 	clusterParams := controllers.CreateClusterParams(&cd, &aci, pullSecret, releaseImageVersion, releaseImageCPUArch, nil)
 
 	if aci.Spec.Networking.NetworkType != "" {
-		clusterParams.NetworkType = &aci.Spec.Networking.NetworkType
+		if aci.Spec.Networking.NetworkType == "OpenshiftSDN" || aci.Spec.Networking.NetworkType == "OVNKubernetes" {
+			clusterParams.NetworkType = &aci.Spec.Networking.NetworkType
+		} else {
+			networkType := "OVNKubernetes"
+			clusterParams.NetworkType = &networkType
+			networkAnnotation := fmt.Sprintf("{\"networking\":{\"networkType\":\"%s\"}}", aci.Spec.Networking.NetworkType)
+			aci.ObjectMeta.SetAnnotations(map[string]string{"InstallConfigOverrides": networkAnnotation})
```

Comment:
It isn't safe to replace the InstallConfigOverrides, you'd have to patch it in.
It's probably better to do that in the installer rather than here.

Reply:
We can do it here https://github.com/openshift/installer/pull/7223